### PR TITLE
Add `bridge.filter.users` config option to configure filtering of direct chats

### LINKED ISSUE #665: Ability to disable direct messages
Hello,

At our server we have bridge to external group (TG) where other members of our project reside. I wanted for our users to be double puppeteer-ed (it's cleaner), so I asked them to login. Unfortunately, it's generating a lot of noise with their direct messages. Is there a way to keep double puppeteering and lose direct messages? I've been looking for solution for quite a long time and found nothing (filters don't matter for direct messages).

I could maybe try to look into code and prepare a PR, but I'm not sure how you want to proceed here.

Thanks in advance

### PR DESCRIPTION
**Pull request by Steffo99**
- If `bridge.filter.users` is "null", direct chats will follow the previous settings.
- If `bridge.filter.users` is "true", direct chats will always be bridged.
- If `bridge.filter.users` is "false", direct chats will never be bridged.

Marked as draft as I did not test it locally.

Fixes #665.

### CODE REVIEW

**Review by tulir — CHANGES_REQUESTED** [review]
The new field also needs to be added to the config upgrader in config.py, otherwise looks fine

**Comment by Steffo99** [comment]
Please test this before merging, I currently cannot test it on my machine due to me not having a development Synapse server (yet)...

File: mautrix_telegram/config.py [hunk 1]
```diff
@@ -197,6 +197,10 @@ def do_update(self, helper: ConfigUpdateHelper) -> None:
 
         copy("bridge.filter.mode")
         copy("bridge.filter.list")
+        if "bridge.filter.users" not in self:
+            base["bridge.filter.users"] = True
+        else:
+            copy("bridge.filter.users")
```
Comment (tulir) [inline mautrix_telegram/config.py:203]: Missing fields are already handled by `copy`

```suggestion
        copy("bridge.filter.users")
```

**Review by tulir — APPROVED** [review]

File: mautrix_telegram/config.py (hunk 1 above)
Comment (Steffo99) [reply mautrix_telegram/config.py]: Does it provide the correct default, though...?

Or does it just set it to `None`?

**Pull request merged**